crypto/x509/x509_vfy.c

```diff
@@ -1673,13 +1673,15 @@ static int check_policy(X509_STORE_CTX *ctx)
         int i;
 
         /* Locate certificates with bad extensions and notify callback. */
-        for (i = 1; i < sk_X509_num(ctx->chain); i++) {
+        for (i = 0; i < sk_X509_num(ctx->chain); i++) {
             X509 *x = sk_X509_value(ctx->chain, i);
 
             CB_FAIL_IF((x->ex_flags & EXFLAG_INVALID_POLICY) != 0,
                        ctx, x, i, X509_V_ERR_INVALID_POLICY_EXTENSION);
         }
-        return 1;
+        /* Should not be able to get here */
+        ERR_raise(ERR_LIB_X509, ERR_R_INTERNAL_ERROR);
+        return 0;
```

Comment:
I am not sure this is correct. The callback can ignore the error and in such case this should not return 0. Shouldn't it?

Reply:
Maybe add some check that we called the callback at least once in this case - this is the _shouldn't happen_ situation.

Reply:
Ah....looks like you're right. I'll create a fixup...

Reply:
Fixed